File: exercism/go/rna-transcription/rna_transcription.go
package strand

func ToRNA(dna string) string {
	rna := make([]byte, len(dna))
	for i := range dna {
		switch dna[i] {
		case 'G':
			rna[i] = 'C'
		case 'C':
			rna[i] = 'G'
		case 'T':
			rna[i] = 'A'
		case 'A':
			rna[i] = 'U'
		}
	}
	return string(rna)
}

File: exercism/zig/rna-transcription/rna_transcription.zig
const mem = @import("std").mem;

pub fn toRna(allocator: mem.Allocator, dna: []const u8) mem.Allocator.Error![]const u8 {
    const rna: []u8 = try allocator.alloc(u8, dna.len);
    for (dna, 0..) |n, i| {
        rna[i] = switch (n) {
            'G' => 'C',
            'C' => 'G',
            'T' => 'A',
            'A' => 'U',
            else => unreachable,
        };
    }
    return rna;
}
